### PR TITLE
Cleanup split gadgets

### PR DESCRIPTION
**Pull request by dkochmanski**
Put gadgets in a separate module. It is a self-contained changed factored out from a local branch for adding themes to frame managers. No functional change is expected – all changes were a typical cleanup without adding or removing interfaces. Only slots from the protocol class gadget has been moved to a basic-gadget.

### CODE REVIEW

File: Core/clim-core/gadgets/concrete.lisp [hunk 1]
```diff
@@ -1872,53 +736,53 @@ and must never be nil.")
...
-                            5 ;(- position slider-button-half-short-dim)
-                            (- middle slider-button-half-long-dim)))))))))
+(declare (ignore region))
+(let ((position (convert-value-to-position pane))
+(slider-button-half-short-dim (ash slider-button-short-dim -1))
+(slider-button-half-long-dim (ash slider-button-long-dim -1)))
+(multiple-value-bind (x1 y1 x2 y2) (bounding-rectangle* (sheet-region pane))
+(display-gadget-background pane (gadget-current-color pane) 0 0 (- x2 x1) (- y2 y1))
+(if (eq (gadget-orientation pane) :vertical)
+            ; vertical case             ; ; ;
```
Comment (lokedhs) [inline Core/clim-core/gadgets/concrete.lisp]: Are these `; ; ;` intentional?

File: Core/clim-core/gadgets/concrete.lisp [hunk 2]
```diff
@@ -1872,53 +736,53 @@ and must never be nil.")
...
++white+
+(- middle slider-button-half-long-dim) (- position slider-button-half-short-dim)
++black+
+(+ middle slider-button-half-long-dim) (+ position slider-button-half-short-dim))
+(when (gadget-show-value-p pane)
+(draw-text* pane (format-value (gadget-value pane)
+(slider-decimal-places pane))
+5 ;(- middle slider-button-half-short-dim)
+10))) ;(- position slider-button-half-long-dim)
+            ; horizontal case           ; ; ;
```
Comment (lokedhs) [inline Core/clim-core/gadgets/concrete.lisp]: Same question as line 746.

File: Core/clim-core/gadgets/concrete.lisp [hunk 3]
```diff
@@ -1879,47 +742,54 @@ and must never be nil.")
...
                                  +black+
-                                 (+ position slider-button-half-short-dim) (+ middle slider-button-half-long-dim))
+                                 (+ position slider-button-half-short-dim)
+                                 (+ middle slider-button-half-long-dim))
               (when (gadget-show-value-p pane)
                 (draw-text* pane (format-value (gadget-value pane)
                                                (slider-decimal-places pane))
                             5 ;(- position slider-button-half-short-dim)
                             (- middle slider-button-half-long-dim)))))))))
-|#
```
Comment (lokedhs) [inline Core/clim-core/gadgets/concrete.lisp]: End-of-comment was removed?

File: Core/clim-core/gadgets/concrete.lisp [hunk 4]
```diff
@@ -2869,12 +1738,12 @@ if INVOKE-CALLBACK is given."))
          (height (space-requirement-height sr)))
     (multiple-value-bind (x y)(output-record-position record)
```
Comment (lokedhs) [inline Core/clim-core/gadgets/concrete.lisp]: Can we add a space after `(x y)` here?

Comment (dkochmanski) [reply Core/clim-core/gadgets/concrete.lisp]: done

File: Core/clim-core/gadgets/concrete.lisp (hunk 3 above)
Comment (dkochmanski) [reply Core/clim-core/gadgets/concrete.lisp]: it was changed to #+(or) for the whole form.

File: Core/clim-core/gadgets/concrete.lisp (hunk 2 above)
Comment (dkochmanski) [reply Core/clim-core/gadgets/concrete.lisp]: fixed (indentation is flawed with #| |#)

File: Core/clim-core/gadgets/concrete.lisp (hunk 1 above)
Comment (dkochmanski) [reply Core/clim-core/gadgets/concrete.lisp]: ditto

**Pull request merged**